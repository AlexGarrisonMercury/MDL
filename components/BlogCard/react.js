import React from 'react'
import classNames from 'classnames/bind'
import PropTypes from 'prop-types'
import style from './style.module.scss'

const cx = classNames.bind(style)

function BlogCard({
  title, content, to, action,
}) {
  return (
    <article className={cx('blog-card')}>
      <a
        href={to}
        onClick={e => {
          e.preventDefault()
          action(to)
        }}
      >
        <div className={cx('blog-card__content')}>
          <h2 className={cx('blog-card__title')}>{title}</h2>
          {content}
        </div>
      </a>
    </article>
  )
}

BlogCard.propTypes = {
  title: PropTypes.string,
  content: PropTypes.string,
  to: PropTypes.string,
  action: PropTypes.func,
}

BlogCard.defaultProps = {
  title: null,
  content: null,
  to: null,
  action: null,
}

export default BlogCard
