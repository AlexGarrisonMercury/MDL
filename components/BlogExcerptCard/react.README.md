# Usage

Used for displaying a quick excerpt and link for a blog post.

<!-- PROPS -->

# Props

title
content
image
link
