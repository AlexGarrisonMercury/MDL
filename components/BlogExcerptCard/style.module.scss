@import 'styles/index';

.card {
  width: 320px;
  height: 375px;
  border: 2px solid $color-m-primary-gray;
  display: flex;
  flex-direction: column;
  font-family: $font-family;

  .image {
    height: 200px !important; // enforce height
    width: 100% !important; // enforce width
    background-repeat: no-repeat;
    background-size: cover;
    background-position: bottom center;
    border-bottom: 2px solid $color-m-primary-gray;
  }

  .content {
    padding: 0.5rem 1.5rem;
    position: relative;
    flex-grow: 1;

    h3 {
      font-size: 18px !important; // enforce mdl size
      color: $color-m-black !important; // enforce mdl size
    }

    p {
      color: $color-m-font-gray;
    }

    a {
      float: right;
      color: $color-m-black;
      font-size: 14px;
      line-height: 1.5rem;
      margin-right: auto;
      text-decoration: none; // overrride other theme classes
      position: relative;

      margin-bottom: 0.5rem;
      padding-top: 0.15rem;

      &:hover {
        color: $color-onyx;
        text-decoration: none;

        &::after {
          background: $color-limerick;
        }
      }

      &::after {
        position: absolute;
        content: '';
        height: 1px;
        /* adjust this to move up and down. you may have to adjust the line height of the paragraph if you move it down a lot. */
        bottom: 0px;
        left: 0;
        width: 100%;
        background: $color-slategray;
      }
    }
  }
}

.active {
  display: block;
}

.inactive {
  display: none;
}
