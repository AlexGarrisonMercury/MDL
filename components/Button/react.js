import React from 'react'
import classNames from 'classnames/bind'
import PropTypes from 'prop-types'
import s from './style.module.scss'

const cx = classNames.bind(s)

const Button = ({
  children, disabled, fullWidth, onClick, variant, size, uppercase, ...props
}) => {
  const classes = cx(['btn'], {
    'btn--primary': variant === 'primary',
    'btn--secondary': ['secondary', 'dark'].includes(variant),
    'btn--default': ['default', 'light'].includes(variant),
    'btn--uppercase': uppercase,
    'btn--sm': size === 'small' || size === 'sm',
    'btn--disabled': disabled,
    'btn--full-width': fullWidth,
  })

  return (
    <button type="button" className={classes} onClick={onClick} disabled={disabled} {...props}>
      {children}
    </button>
  )
}

Button.propTypes = {
  children: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  variant: PropTypes.oneOf(['default', 'primary', 'secondary']),
  size: PropTypes.oneOf(['small', 'sm']),
  uppercase: PropTypes.bool,
  fullWidth: PropTypes.bool,
}

Button.defaultProps = {
  disabled: false,
  onClick: null,
  variant: 'default',
  size: null,
  uppercase: false,
  fullWidth: false,
}

export default Button
