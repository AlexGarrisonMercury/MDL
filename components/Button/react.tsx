import React, { FC, ReactNode } from 'react'
import classNames from 'classnames/bind'
import { mCheck } from '@masonite/svg-icons'
import s from './style.module.scss'
import Icon from '../Icon/react'

const cx = classNames.bind(s)

export interface ButtonProps {
  children: ReactNode
  /** Adds CSS classes after all computed button classes. */
  className?: string
  disabled?: boolean
  /** Sets the button to take up 100% width of its parent. */
  fullWidth?: boolean
  onClick?: any
  /** The button style. If none is provided, the button background is transparent with no shadow. */
  variant?: 'default' | 'primary' | 'secondary' | 'light'
  size?: 'sm' | 'lg' | 'small' | 'large'
  /** Apply inline styles directly to the button. */
  style?: object
  uppercase?: boolean
  active?: boolean
  /** Makes the text bold. The default style is already bold. */
  bold?: boolean
}

const Button: FC<ButtonProps> = ({
  children,
  className,
  disabled,
  fullWidth,
  onClick,
  variant,
  size,
  style,
  uppercase,
  active,
  bold,
  ...props
}) => {
  const classes = cx(
    'btn',
    {
      'btn--primary': variant === 'primary',
      'btn--secondary': ['secondary', 'dark'].includes(variant),
      'btn--light': variant === 'light',
      'btn--default': variant === 'default',
      'btn--uppercase': uppercase,
      'btn--sm': size === 'small' || size === 'sm',
      'btn--large': size === 'large' || size === 'lg',
      'btn--disabled': disabled,
      'btn--full-width': fullWidth,
      'btn--bold': bold,
      'btn--active': active,
      'btn--large-active': (size === 'large' || size === 'lg') && active,
      'btn--small-active': (size === 'small' || size === 'sm') && active,
    },
    className,
  )

  return (
    <button
      type="button"
      className={classes}
      onClick={onClick}
      disabled={disabled}
      style={style}
      {...props}
    >
      {children}
      {active && size === 'large' && (
        <span className={cx('btn--active-check')}>
          <Icon svg={mCheck} size={16} />
        </span>
      )}
    </button>
  )
}

export default Button
