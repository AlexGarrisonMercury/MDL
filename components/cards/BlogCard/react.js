import React, { Component } from 'react'
import classNames from 'classnames/bind'
import style from './style.module.scss'

const cx = classNames.bind(style)

export default function BlogCard(props) {
  const {
    title, content, image, to,
  } = props

  return (
    <article className={cx('blog-card')}>
      <a
        href={to}
        onClick={e => {
          e.preventDefault()
          action(to)
        }}
      >
        <div className={cx('blog-card__content')}>
          <h2 className={cx('blog-card__title')}>{title}</h2>
          {content}
        </div>
      </a>
    </article>
  )
}
