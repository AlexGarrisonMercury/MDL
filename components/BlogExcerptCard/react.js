import React from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';
import style from './style.module.scss';
// components
import { Heading, P } from '../typography/react';

const cx = classNames.bind(style);

const BlogExcerptCard = ({ title, content, image, link }) => {
  return (
    <div className={classes.card}>
      <div className={classes.image} style={{ backgroundImage: `url(${image})` }} />
      <div className={classes.content}>
        <Heading level='3'>{title}</Heading>
        <P>{content}</P>
        <a href={link}>Share</a>
      </div>
    </div>
  );
};

BlogExcerptCard.propTypes = {
  /**
   * The card title.
   */
  title: PropTypes.string.isRequired,
  /**
   * The card's content area.
   */
  content: PropTypes.string.isRequired,
  /**
   * Image url used on the card
   */
  image: PropTypes.string.isRequired,
  /**
   * Href value for share button
   */
  link: PropTypes.string.isRequired,
};

BlogExcerptCard.defaultProps = {
  title: null,
  content: null,
  image: null,
  link: null,
};

export default BlogExcerptCard;
