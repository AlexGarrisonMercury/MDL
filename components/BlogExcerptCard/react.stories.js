import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';
import BlogExcerptCard from './react';
import Readme from './react.README.md';

storiesOf('BlogExcerptCard', module)
  .addParameters({ readme: { sidebar: Readme } })
  .addDecorator(withKnobs)
  .add('Default', () => (
    <BlogExcerptCard
      title='Why Stile and Rail Doors are Worth the Investment'
      content='When you want the best quality wood doors for a hospitality or office project, stile and rail doors provide both...'
      image='https://embed.widencdn.net/img/masonite/8hui9fjtwo/exact/Harring_Hotel%20Indigo_2_L2.jpeg'
      link='#'
    />
  ));
