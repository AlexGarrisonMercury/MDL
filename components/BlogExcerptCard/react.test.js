import React from 'react';
import { render } from '@testing-library/react';
import BlogExcerptCard from './react';

describe('BlogExcerptCard', () => {
  it('should render without crashing', () => {
    const { asFragment } = render(
      <BlogExcerptCard
        title='Why Stile and Rail Doors are Worth the Investment'
        content='When you want the best quality wood doors for a hospitality or office project, stile and rail doors provide both...'
        image='https://embed.widencdn.net/img/masonite/8hui9fjtwo/exact/Harring_Hotel%20Indigo_2_L2.jpeg'
        link='#'
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
