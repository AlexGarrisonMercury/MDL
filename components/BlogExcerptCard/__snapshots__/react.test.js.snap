// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogExcerptCard should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="image"
      style="background-image: url(https://embed.widencdn.net/img/masonite/8hui9fjtwo/exact/Harring_Hotel%20Indigo_2_L2.jpeg);"
    />
    <div
      class="content"
    >
      <h3
        class="h3"
      >
        Why Stile and Rail Doors are Worth the Investment
      </h3>
      <p
        class="p"
      >
        When you want the best quality wood doors for a hospitality or office project, stile and rail doors provide both...
      </p>
      <a
        href="#"
      >
        Share
      </a>
    </div>
  </div>
</DocumentFragment>
`;
